Runtime/Core/ShrinkablePolygon.cs

```diff
@@ -0,0 +1,1102 @@
+                            m_CantIntersect = true,
+                            m_OriginalPosition = ShrinkablePoint2.m_Vector2NaN,
+                        };
+                        
+                        int nextIndex = (i == m_Points.Count - 1 ? 0 : i + 1);
+                        extendedPoints[i * 3 + 2] = new ShrinkablePoint2
+                        {
+                            m_Position = Vector2.Lerp(m_Points[i].m_Position, m_Points[nextIndex].m_Position, 0.01f),
+                            m_ShrinkDirection = m_Points[i].m_ShrinkDirection,
+                            m_CantIntersect = true,
+                            m_OriginalPosition = ShrinkablePoint2.m_Vector2NaN,
+                        };
+                    }
+                }
+                
+                // TODO: remove unused
+                // remove unused padding
+                for (int index = extendedPoints.Count - 1; index >= 0; index--)
+                {
+                    if (extendedPoints[index].m_Position == Vector2.zero &&
+                        extendedPoints[index].m_OriginalPosition == Vector2.zero &&
+                        extendedPoints[index].m_ShrinkDirection == Vector2.zero
+                        )
+                    {
+                        extendedPoints.RemoveAt(index);
+                    }
+                }
+
+                m_Points = extendedPoints;
+            }
+        }
+    
+        private static readonly List<Vector2> m_cachedShrinkDirections = new List<Vector2>(100); 
+        /// <summary>
+        /// Computes shrink directions that respect the aspect ratio of the camera. If the camera window is a square,
+        /// then the shrink directions will be equivalent to the normals.
+        /// </summary>
+        public void ComputeAspectBasedShrinkDirections()
+        {
+            // cache current shrink directions to check for change later
+            m_cachedShrinkDirections.Clear();
+            for (int i = 0; i < m_Points.Count; ++i)
+            {
+                m_cachedShrinkDirections.Add(m_Points[i].m_ShrinkDirection);
+            }
+            
+            // calculate shrink directions
+            ComputeNormals(false);
+            for (int i = 0; i < m_Points.Count; ++i)
+            {
+                int prevIndex = i == 0 ? m_Points.Count - 1 : i - 1;
+                int nextIndex = i == m_Points.Count - 1 ? 0 : i + 1;
+
+                var mPoint = m_Points[i];
+                mPoint.m_ShrinkDirection = CalculateShrinkDirection(mPoint.m_ShrinkDirection, 
+                    m_Points[prevIndex].m_Position, mPoint.m_Position, m_Points[nextIndex].m_Position);
+                m_Points[i] = mPoint;
+            }
+
+            // update m_State, if change happened based on the cached shrink directions
+            if (m_cachedShrinkDirections.Count != m_Points.Count)
+            {
+                m_State++; // m_State change if more points where added
```

Comment:
this can never happen because you passed false to ComputeNormals()

Reply:
Good catch. It's a remnant from a previous life where there was no option to pass false :)